components/script/animations.rs

```diff
@@ -488,6 +512,8 @@ impl Animations {
                     .fire(node.upcast());
             }
         }
+
+        self.update_running_animations_presence(window, None);
```

Comment:
With the above change this would become:

```rust
        if self.pending_events.borrow().is_empty() {
            self.handle_animation_presence_or_pending_events_change(window);
        }
```

This requires an early return at the top if `events.is_empty()`.